test/app/users/getManageConsoleServices.test.js

```diff
@@ -0,0 +1,183 @@
+jest.mock('./../../../src/app/users/utils');
+
+const getManageConsoleServices = require('./../../../src/app/users/getManageConsoleServices');
+const { sendResult } = require('./../../../src/infrastructure/utils');
+const { getAllServices } = require('./../../../src/infrastructure/applications');
+const { getServicesByUserId } = require('./../../../src/infrastructure/access');
+const { getUserDetails } = require('./../../../src/app/users/utils');
+
+describe('When retrieving manage console services for a user', () => {
+  let req;
+  let res;
+
+  beforeEach(() => {
+    req = {
+      id: 'correlationId',
+      csrfToken: () => 'token',
+      accepts: () => ['text/html'],
+      user: {
+        sub: 'user1',
+        email: 'super.user@unit.test',
+      },
+      params: {
+        uid: 'user1',
+      },
+      session: {},
+    };
+
+    res = {
+      render: jest.fn(),
+    };
+
+    getUserDetails.mockReset();
+    getUserDetails.mockReturnValue({
+      id: 'user1',
+    });
+
+    getServicesByUserId.mockReset();
+    getServicesByUserId.mockReturnValue([
+      {
+        userId: "user-1",
+        serviceId: "service1Id",
+        organisationId: "organisation-1",
+        roles: [],
+      },
+      {
+        userId: "user-1",
+        serviceId: "service2Id",
+        organisationId: "organisation-1",
+        roles: [],
+      },
+      {
+        userId: "user-1",
+        serviceId: "service3Id",
+        organisationId: "organisation-1",
+        roles: [],
+      },
+      {
+        userId: "user-1",
+        serviceId: "service4Id",
+        organisationId: "organisation-1",
+        roles: [],
+      },
+    ]);
+
+
+    const allServices = {
+      services: [
+        {
+          id: 'service1Id',
+          name: 'Service 1',
+          description: 'Service for testing purposes',
+          isExternalService: true,
+          isIdOnlyService: false,
+          isHiddenService: false,
+          isMigrated: false,
+          relyingParty: {},
+        },
+        {
+          id: 'service2Id',
+          name: 'Service 2',
+          description: 'Service for testing purposes',
+          isExternalService: true,
+          isIdOnlyService: false,
+          isHiddenService: false,
+          isMigrated: false,
+          relyingParty: {},
+        },
+        {
+          id: 'service3Id',
+          name: 'Service 3',
+          description: 'Service for testing purposes',
+          isExternalService: true,
+          isIdOnlyService: false,
+          isHiddenService: false,
+          isMigrated: false,
+          relyingParty: {},
+        },
+        {
+          id: 'service4Id',
+          name: 'Service 4',
+          description: 'Service for testing purposes',
+          isExternalService: true,
+          isIdOnlyService: false,
+          isHiddenService: false,
+          isMigrated: false,
+          relyingParty: {},
+        },
+        {
+          id: 'service5Id',

```

Comment:
The parameters of this test service seem the same as the one above; is it necessary to have this many?

Reply:
Yep, totally. I think the test did need it at one point but ended up simplifying it.